src/flb_io.c

```diff
@@ -102,6 +102,13 @@ int flb_io_net_connect(struct flb_upstream_conn *u_conn,
                   u_conn->fd, u->tcp_host, u->tcp_port);
     }
 
+    /* set TCP keepalive if configured */
+    ret = flb_net_socket_tcp_keepalive(fd, &u->net);
```

Comment:
this line is changing the default system values, I think this call must be optional based on if the user set some parameters. I mean: all the internal tcp_something fields should have a non-set value like ```-1``` and IF in the invoked call that value is >= 0, then perform the syscall to change the behavior.

Reply:
changed. Added a boolean: `net.tcp_keepalive`, that allows enabling keepalives while keeping the system-wide default values